### DISCUSSION TITLE
q-file BROKEN in Safari on Mac / iOS (Vue 2, Quasar 1.15.10) - workaround?

### DISCUSSION THREAD

**Discussion by madLinux7** [quasarframework/quasar#11094] [General - Components / Directives / etc] [answered]
I discovered that in Safari or a capacitor build on iOS (only on physical devices, not on simulator... weird enough) you cannot make more than one attempt to add files to a plain q-file component with the 'multiple' attribute. When you try it, depending on the way you configured your component, you either get `[Error] [Vue warn]: Duplicate keys detected: '{FILE}'. This may cause an update error.` consistently logged in the console or just a silent crash - in the end the page stops responding.

This is my q-file component:

`<q-file append multiple ref="test" name="test" for="test" label="test" v-model="tileUploaderFiles" />`

Im also referring to my latest issuing here on GitHub where i get these logs: https://github.com/quasarframework/quasar/discussions/11071

Is there some kind of workaround for this in Safari / iOS safari web-container? I just need to store these file infos for a later upload to be issued, but im unsure what would be the best approach.

Any help is appreciated!!

**Comment by ontwikkelfabriek** [comment]
@YourPenguin 

Do you know if this bug also effects Quasar v2?
This looks very broken. Please create a GitHub issue for this.

**Comment by hawkeye64** [comment, marked as answer]
@YourPenguin I gave an answer to  #11071. I think this is very similar. Be careful with Vue `key` when in `v-for` loops.
If this is not the case, then provide a CodePen to demonstrate the issue.

**Reply by madLinux7** [reply]
Thanks for this reply. In the process of creating a CodePen i realized quasar v1.16.0 is used in the CodePen example while i was still on 1.15.11... such a banal solution, but now everything seems to work fine :D
